compiler/main.nim

```diff
@@ -63,9 +63,7 @@ proc commandCheck(graph: ModuleGraph) =
 
 when not defined(leanCompiler):
   proc commandDoc2(graph: ModuleGraph; json: bool) =
-    if optWholeProject in graph.config.globalOptions:
-      # Backward compatibility with previous versions
-      graph.config.outDir = AbsoluteDir(graph.config.outDir / graph.config.outFile)
```

Comment:
I've checked this code out again. I believe the better way is to re-add this logic to `getOutFile2` in `docgen.nim` instead of yet another `proc`.

Reply:
Ok, but I'll do it later.